components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/tests/Services/GetSimpleEventFilterTests.cs

```diff
@@ -0,0 +1,288 @@
+            var operand = (LiteralOperand)eventFilter.WhereClause.Elements[0].FilterOperands[0].Body;
+            Assert.IsType<NodeId>(operand.Value.Value);
+            var nodeId = (NodeId)operand.Value.Value;
+            Assert.Equal(nodeId, ObjectTypeIds.ConditionType);
+        }
+
+        protected override Mock<INodeCache> SetupMockedNodeCache() {
+            var nodeCache = base.SetupMockedNodeCache();
+            AddNode(_baseObjectTypeNode);
+            AddNode(_baseEventTypeNode);
+            AddNode(_messageNode);
+            AddNode(_conditionTypeNode);
+            AddNode(_conditionNameNode);
+            AddNode(_commentNode);
+            AddNode(_enabledStateNode);
+            AddNode(_idNode);
+            var typeTable = nodeCache.Object.TypeTree as TypeTable;
+            typeTable.Add(_baseObjectTypeNode);
+            typeTable.Add(_baseEventTypeNode);
+            typeTable.AddSubtype(ObjectTypeIds.BaseEventType, ObjectTypeIds.BaseObjectType);
+            typeTable.AddSubtype(ObjectTypeIds.ConditionType, ObjectTypeIds.BaseEventType);
+            _baseObjectTypeNode.ReferenceTable.Add(ReferenceTypeIds.HasSubtype, false, ObjectTypeIds.BaseEventType);
+            _baseEventTypeNode.ReferenceTable.Add(ReferenceTypeIds.HasSubtype, true, ObjectTypeIds.BaseObjectType);
+            _baseEventTypeNode.ReferenceTable.Add(ReferenceTypeIds.HasProperty, false, _messageNode.NodeId);
+            _messageNode.ReferenceTable.Add(ReferenceTypeIds.HasProperty, true, ObjectTypeIds.BaseEventType);
+            _baseEventTypeNode.ReferenceTable.Add(ReferenceTypeIds.HasSubtype, false, ObjectTypeIds.ConditionType);
+            _conditionTypeNode.ReferenceTable.Add(ReferenceTypeIds.HasSubtype, true, ObjectTypeIds.BaseEventType);
+            _conditionTypeNode.ReferenceTable.Add(ReferenceTypeIds.HasProperty, false, _conditionNameNode.NodeId);
+            _conditionTypeNode.ReferenceTable.Add(ReferenceTypeIds.HasProperty, false, _commentNode.NodeId);
+            _conditionNameNode.ReferenceTable.Add(ReferenceTypeIds.HasProperty, true, ObjectTypeIds.ConditionType);
+            _conditionTypeNode.ReferenceTable.Add(ReferenceTypeIds.HasComponent, false, _enabledStateNode.NodeId);
+            _enabledStateNode.ReferenceTable.Add(ReferenceTypeIds.HasComponent, true, ObjectTypeIds.ConditionType);
+            _enabledStateNode.ReferenceTable.Add(ReferenceTypeIds.HasProperty, false, _idNode.NodeId);
+            _idNode.ReferenceTable.Add(ReferenceTypeIds.HasProperty, true, _enabledStateNode.NodeId);
+            _commentNode.ReferenceTable.Add(ReferenceTypeIds.HasProperty, true, ObjectTypeIds.ConditionType);
+            nodeCache.Setup<Node>(x => x.FetchNode(It.IsAny<ExpandedNodeId>())).Returns((ExpandedNodeId x) => {
+                if (x.IdType == IdType.Numeric && x.Identifier is uint id) {
+                    return _nodes[id];
+                }
+                else {
```

Comment:
Detail: You don't need the else statement here.

Reply:
Fixed